src/chains/ethereum/console.log/src/handlers.ts

```diff
@@ -32,27 +32,29 @@ const OR_WITH_TWOS_COMPLEMENT =
  * @returns The starting position of the value in `memory`, and the `end` position of the value.
  */
 function getDynamicDataMarkers(memory: Buffer, offset: number) {
-  // A Buffer's length maxes out at UInt32 max, 4294967295, (2**(8 * 4)) - 1) it
-  // is safe to decode the `start` and `length` values as UInt32s.
+  // A Buffer's length maxes out at UInt32 max, 4294967295, (2**(8 * 4)) - 1) so
+  // it is safe to decode the `start` and `length` values as UInt32s.
   //
   // If we had a start position "word" of:
   //   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 60
   // we know we really only need to read the last 4 bytes, since we know that
   // a Buffer's length can't be greater than UInt32 (which can be represented by
   // up to 4 bytes). So we only need to look at:
   // 00 00 00 60
-  // `offset + WORD_SIZE - 4` effectively tells `readUInt32BE` to read only
-  // these last 4 bytes of the word that represent the `start`
+  // `offset + WORD_SIZE - UINT32_BYTE_LENGTH` effectively tells `readUInt32BE`
+  // to read only these last 4 bytes of the word that represent the `start`
   // value.
-  // We add WORD_SIZE to this value because without it it doesn't work.
-  // We add `4` because `memory` includes the function signature, which is 4
-  // bytes, but the `start` position doesn't.
-  const start = memory.readUInt32BE(offset + WORD_SIZE - 4) + WORD_SIZE + 4;
+  // We add WORD_SIZE to this value because we are skipping over the location of
+  // the length and referencing the start of the location of the data itself.
+  const UINT32_BYTE_LENGTH = 4;
+  const nextWordOffset = offset + WORD_SIZE;
+  const startUint32MemoryPosition = nextWordOffset - UINT32_BYTE_LENGTH;
+  const start = memory.readUInt32BE(startUint32MemoryPosition) + WORD_SIZE;
```

Comment:
```suggestion
  const UINT32_BYTE_LENGTH = 4;
  const nextWordOffset = offset + WORD_SIZE;
  const startUint32MemoryPosition = nextWordOffset - UINT32_BYTE_LENGTH;
  // We add WORD_SIZE to this value because we are skipping over the location of
  // the length and referencing the start of the location of the data itself.
  const start = memory.readUInt32BE(startUint32MemoryPosition) + WORD_SIZE;
```
Moving comment to be over the line it's actually referencing

Reply:
If this line is moved I think the comment above that needs to be reworded and moved. The whole comment block describes what will happen and why; it currently isn't "anchored" to any of the code lines.

Reply:
Ahh, I see your point. Either is fine with me